### ISSUE TITLE
ShipLocker.json failed to decode

### ISSUE THREAD

**Issue by norohind** [EDCD/EDMarketConnector#1216]
**Please complete the following information:**
 - Version: 5.1.2+066bd60
 - Game Version: Odyssey
 - OS: Windows 10
 - OS Locale: Russian
[EDMarketConnector-debug-5.1.2_02.log](https://github.com/EDCD/EDMarketConnector/files/6932295/EDMarketConnector-debug-5.1.2_02.log)
[5.1.2_02.log](https://github.com/EDCD/EDMarketConnector/files/6932297/5.1.2_02.log)

**Describe the bug**
EDMC couldn't parse ShipLocker.json and put it to logs
```
2021-08-04 17:21:23.199 - WARNING - 42528:35240:35240 monitor.EDLogs.parse_entry:873: ShipLocker.json failed to decode:
JSONDecodeError('Expecting value: line 1 column 1 (char 0)')
```

**To Reproduce**
1. Manage to trigger ShipLocker.json parsing
2. Hope it will fail

**Expected behavior**
EDMC successfully parse ShipLocker.json in all cases

**Screenshots**
None

**Additional context**
I got this only one time

**Comment by A-UNDERSCORE-D** [comment]
Note to self: Race condition between write of journal ShipLocker (and others, it affects all of those) and our read of the discrete files. Suggested fix is a 10ms sleep if this exact exception is hit.

**Comment by Athanasius** [comment]
> Note to self: Race condition between write of journal ShipLocker (and others, it affects all of those) and our read of the discrete files. Suggested fix is a 10ms sleep if this exact exception is hit.

Or, on exception, check the actual size of the file, although that might hit the "slow metadata update" issue on Windows.

Mostly I'd like to not have only a *single* retry, rather than introducing an extra layer of race.

**Comment by A-UNDERSCORE-D** [comment]
possibly a loop trying 5 times? small sleep between, something like that

**Comment by Athanasius** [comment]
Fix is in 5.1.3

**Comment by Kawazu0** [comment]
Hello,
I have this issue in latest EDMC release. (updated with git)
Excerpt of "journal" (I'm using python version under Linux with Steam running Elite under proton)
```
2023-08-19 20:25:29.008 UTC - INFO - 6682:140593494925376:6682 EDMarketConnector.__main__:2114: Startup v5.9.4+6e8f8815 : Running on Python v3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
2023-08-19 20:25:29.010 UTC - INFO - 6682:140593494925376:6682 protocol.LinuxProtocolHandler.__init__:382: Web server listening on http://localhost:45605/auth
E:D Market Connector 5.9.4+6e8f8815
```
[.... Cut many messages without error ...]
`2023-08-19 22:06:52.412 UTC - ERROR - 6682:140593494925376:6682 monitor.EDLogs.parse_entry:1437: LoadoutEquipModule: OrderedDict([('timestamp', '2023-08-19T22:06:52Z'), ('event', 'LoadoutEquipModule'), ('LoadoutName', 'artemis 3'), ('SuitID', 1751960305440585), ('SuitName', 'explorationsuit_class3'), ('SuitName_Localised', '$ExplorationSuit_Class1_Name;'), ('LoadoutID', 4293000003), ('SlotName', 'PrimaryWeapon1'), ('ModuleName', 'wpn_m_submachinegun_kinetic_fauto'), ('ModuleName_Localised', 'Karma C-44'), ('Class', 3), ('WeaponMods', ['weapon_range', 'weapon_clipsize']), ('SuitModuleID', 1774512138391407)])`
[ some success messages ]
```
2023-08-19 23:01:43.280 UTC - WARNING - 6682:140593494925376:6682 monitor.EDLogs.parse_entry:1123: ShipLocker.json failed to decode:
JSONDecodeError('Expecting value: line 1 column 1 (char 0)')
```
[multiple times]
[ some success messages then closing messages ]
```
2023-08-19 23:17:33.360 UTC - INFO - 6682:140593494925376:6682 EDMarketConnector.AppWindow.onexit:1994: Destroying app window...
2023-08-19 23:17:33.413 UTC - INFO - 6682:140593494925376:6682 EDMarketConnector.AppWindow.onexit:1997: Done.
2023-08-19 23:17:33.413 UTC - INFO - 6682:140593494925376:6682 EDMarketConnector.__main__:2306: Exiting
```

During the session, I sold an item to a bartender, went on exploration changed weapons on my Artemis Suit and scanned some exobiology and geology entities. If you need some other details, or if I can be of any help, feel free to ask.

regards
